test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,399 @@
+        svg = TestMethods.generateSVG();
+        plot = new Plottable.Plots.Scatter<number, number>();
+        plot.x((d: any) => d.x)
+            .y((d: any) => d.y);
+      });
+
+      it.skip("correctly handles NaN, undefined, Infinity, and non-number x and y values", () => {
+        let data = [
+          { x: 0.0, y: 0.0 },
+          { x: 0.2, y: 0.2 },
+          { x: 0.4, y: 0.4 },
+          { x: 0.6, y: 0.6 },
+          { x: 0.8, y: 0.8 }
+        ];
+        let dataset = new Plottable.Dataset(data);
+        plot.addDataset(dataset);
+        plot.renderTo(svg);
+
+        assert.strictEqual(plot.selections().size(), 5, "all 5 data points are drawn");
+
+        let dataWithError = data.slice();
+        dataWithError[2] = { x: 0.4, y: NaN };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is NaN");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned  when one of the y values is NaN");
+
+        dataWithError = data.slice();
+        dataWithError[2] = { x: NaN, y: 0.4 };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is NaN");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is NaN");
+
+        dataWithError = data.slice();
+        dataWithError[2] = { x: 0.4, y: undefined };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is undefined");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the y values is undefined");
+
+        dataWithError = data.slice();
+        dataWithError[2] = { x: undefined, y: 0.4 };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is undefined");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is undefined");
+
+        dataWithError = data.slice();
+        dataWithError[2] = { x: 0.4, y: Infinity };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is Infinity");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the y values is Infinity");
+
+        dataWithError = data.slice();
+        dataWithError[2] = { x: -Infinity, y: 0.4 };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is -Infinity");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is -Infinity");
+
+        dataWithError = data.slice();
+        (<any> dataWithError)[2] = { x: 0.4, y: "12" };
```

Comment:
Sorry, now that I look back I believe my comments were confusing and indeed contradictory. Let me try again.

Initially I was suggesting to reduce the test size by just creating an initial object with
```typescript
x = [badValue, badValue, badValue, badValue, goodValue].
assert.lengthOf(plot.entities(), 1) // just the goodValue
```

That meant that the array does not need to be typed.
But then I deleted that comment as I realized this is such a better way of testing it. However, I did not realize that now typing the object is required.
My mistake, I apologize for the contradictory comments. My current stand right now is: we should either type it as `any[]` or `{[key: string]: any }[] `. This is because <any> casting it does not make sense on this line, we want to avoid that.

Convention wise:
- Wherever the type can be inferred, don't type it explicitly 
- Avoid <any> casting as much as possible